src/main/java/io/strimzi/kafka/bridge/mqtt/core/MqttServerHandler.java

```diff
@@ -85,6 +85,14 @@ public void channelActive(ChannelHandlerContext ctx) {
 
     @Override
     protected void channelRead0(ChannelHandlerContext ctx, MqttMessage msg) {
+        if (msg.decoderResult() != null) {
+            Throwable error = msg.decoderResult().cause();
+            if (error != null) {
+                LOGGER.error("Received invalid message with error: {}", error.getMessage());
+                exceptionCaught(ctx, error);
+                return;
+            }
+        }
```

Comment:
From what I understand `decoderResult()` doesn't ever return null? And if the `decoderResult()` is a failure it should have a non-null cause(). If my understanding is correct, we may be able to refactor this like the following:
```suggestion
        if (msg.decoderResult().isFailure()) {
            Throwable error = msg.decoderResult().cause();
            LOGGER.error("Received invalid message with error: {}", error.getMessage());
            exceptionCaught(ctx, error);
            return;
        }
```
What do you think?

Reply:
+1 to Kyle. AFAICS, when an MqttMessage is created as a valid message the decoder result is SUCCESS so it's never empty. See here https://github.com/netty/netty/blob/netty-4.1.118.Final/codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttMessage.java#L52